### PR TITLE
Simplify Jest config a little bit

### PR DESCRIPTION
**Pull request by gaearon**
See individual commits.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,8 +110,6 @@
   },
   "jest": {
     "modulePathIgnorePatterns": [
-      "/.module-cache/",
```
Comment (gaearon) [inline package.json]: This was relevant for Jest 0.1.5 which was ages ago, and @zpao wasn't sure if it was necessary. Let's live without it and see how it goes.

File: package.json [hunk 2]
```diff
@@ -110,8 +110,6 @@
...
-      "<rootDir>/build/",
```
Comment (gaearon) [inline package.json]: This was relevant when we had default root folder, and thus needed to exclude build explicitly. We don't anymore.

File: package.json [hunk 3]
```diff
@@ -132,16 +130,12 @@
       "ts"
     ],
     "roots": [
-      "<rootDir>/eslint-rules",
       "<rootDir>/mocks",
       "<rootDir>/scripts",
-      "<rootDir>/src",
-      "node_modules/fbjs"
```
Comment (gaearon) [inline package.json]: This was relevant when we imported invariant/warning etc directly. Also the reason why tests don't currently fail in this case (but build does).

File: package.json [hunk 4]
```diff
@@ -132,16 +130,12 @@
...
+      "<rootDir>/src"
     ],
     "collectCoverageFrom": [
-      "src/**/*.js",
-      "!src/__mocks__/vendor/third_party/*.js",
```
Comment (gaearon) [inline package.json]: I deleted the file that this was for a few weeks ago.

File: scripts/eslint-rules/__tests__/no-primitive-constructors-test.js [hunk 5]
```diff
@@ -14,11 +14,7 @@ var RuleTester = require('eslint').RuleTester;
 var ruleTester = new RuleTester();
 
 ruleTester.run('eslint-rules/no-primitive-constructors', rule, {
-  valid: [
```
Comment (gaearon) [inline scripts/eslint-rules/__tests__/no-primitive-constructors-test.js]: This is Prettier. (The file was outside of its reach before.)

File: src/renderers/dom/shared/utils/__tests__/getNodeForCharacterOffset-test.js [hunk 6]
```diff
@@ -21,6 +19,17 @@ function createNode(html) {
   return node;
 }
 
+function getTestDocument(markup) {
```
Comment (gaearon) [inline src/renderers/dom/shared/utils/__tests__/getNodeForCharacterOffset-test.js]: I think it's no big deal that we duplicate this. Actually convenient that you directly see what's going on now.

Comment (bvaughn) [reply src/renderers/dom/shared/utils/__tests__/getNodeForCharacterOffset-test.js]: I'm not sure I understand the motivation here, or why it's advantageous to copy/paste this boilerplate into several tests.

**Comment by gaearon** [comment]
Mostly I want to get rid of the only case where we share a module between tests without actually using it. It's confusing to have `src/test` directory with one file. (People generally assume that `src/test` would contain tests for React itself.)

I agree we should have a way to share modules between tests but I want to think about that after reorganizing the project structure. For now I just want to get rid of that single special case since it's one less thing to think about when reorganizing the project.

**Comment by gaearon** [comment]
To be clear, the only way this file was special is because it was shared between tests, but not used in React itself. It was the only file that matches this criteria—perhaps in part because it is not obvious at all that `src/test` is where you’re supposed to put such files.

**Comment by bvaughn** [comment]
This makes much better sense after the explanation. Thanks!

**Pull request merged**